portal-impl/src/com/liferay/portal/util/PortalImpl.java

```diff
@@ -923,15 +921,22 @@ public String escapeRedirect(String url) {
 				else if (allowedDomain.equals(domain)) {
 					return url;
 				}
-				else if (allowedDomain.equals("PORTAL_DOMAIN")) {
-					ServiceContext serviceContext =
-						ServiceContextThreadLocal.getServiceContext();
-
-					ThemeDisplay themeDisplay =
-						serviceContext.getThemeDisplay();
+				else if (allowedDomain.equals("PORTAL_DOMAINS")) {
+					try {
+						if (Validator.isNotNull(
+								CompanyLocalServiceUtil.getCompanyByVirtualHost(
+									domain))) {
```

Comment:
It is best to avoid using exceptions for control flow. It is an anti-pattern. Instead use `CompanyLocalServiceUtil#fetchCompanyByVirtualHost` and check for null. (if you feel debug the logging is helpful).